Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/TransformComponent.cpp

```diff
@@ -954,6 +955,13 @@ namespace AzToolsFramework
                 return AZ::Failure(AZStd::string("You cannot set an entity to be a child of a closed container!"));
             }
 
+            // Don't allow entities to be parented outside their container.
+            AzToolsFramework::EntityIdList entities = m_focusModeInterface->GetFocusedEntities(GetEntityContextId());
+            if (std::find(entities.begin(), entities.end(), actualValue) == entities.end())
+            {
+                return AZ::Failure(AZStd::string("You cannot set an entity to be a child of a different container!"));
+            }
```

Comment:
nit: This works and shouldn't cause performance concerns (the focused entities list is cached and passed by const ref), but alternatively you can also use IsInFocusSubTree (which just returns true/false)

Reply:
Much nicer!